testssl.sh

```diff
@@ -641,6 +641,38 @@ else
 	}
 fi
 
+# arg1: An ASCII-HEX string
+# arg2: file name
+# Append $arg1 in binary format to $arg2
+asciihex_to_binary_file(){
+     local string="$1"
+     local file="$2"
+     local -i len
+     local -i i ip2 ip4 ip6 ip8 ip10 ip12 ip14
+     local -i remainder
+
+     len=${#string}
+     [[ $len%2 -ne 0 ]] && return 1
+
+     for (( i=0; i <= len-16 ; i=i+16 )); do
+          ip2=$i+2; ip4=$i+4; ip6=$i+6; ip8=$i+8; ip10=$i+10; ip12=$i+12; ip14=$i+14
+          echo -e -n "\x${string:i:2}\x${string:ip2:2}\x${string:ip4:2}\x${string:ip6:2}\x${string:ip8:2}\x${string:ip10:2}\x${string:ip12:2}\x${string:ip14:2}" >> "$file"
+     done
+
+     ip2=$i+2; ip4=$i+4; ip6=$i+6; ip8=$i+8; ip10=$i+10; ip12=$i+12; ip14=$i+14
+     remainder=$len-$i
+     case $remainder in
+           2) echo -e -n "\x${string:i:2}" >> "$file" ;;
```

Comment:
Thank you. 

Two points: 
* printf would be more portable instead of echo -n -e
* asciihex_to_binary_file() doesn't look so elegant to me. That's for parsing the certificate?